app/router.go

```diff
@@ -133,8 +133,15 @@ func RegisterReportPostHandler(a Adder, router *mux.Router) {
 
 		decoder := gob.NewDecoder(reader).Decode
 		if strings.HasPrefix(r.Header.Get("Content-Type"), "application/json") {
-			decoder = json.NewDecoder(reader).Decode
+			decoder = func(i interface{}) error {
+				return codec.NewDecoder(reader, &codec.JsonHandle{}).Decode(i)
+			}
```

Comment:
No need for the lambda here (and below):
`decoder = codec.NewDecoder(reader, &codec.JsonHandle{}).Decode`